From d75e1f837e766f54bd1a12e40e41d01167dc176e Mon Sep 17 00:00:00 2001
From: Bart Van Assche <bvanassche@google.com>
Date: Fri, 12 Sep 2025 12:21:43 -0700
Subject: [PATCH 2/2] ANDROID: dm-default-key: Set .flush_bypasses_map

default_key_map() does not modify REQ_PREFLUSH bios. Hence, it is safe
to enable dm-core flush bypassing for this target.
---
 drivers/md/dm-default-key.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/md/dm-default-key.c b/drivers/md/dm-default-key.c
index 5f379ab70d20..e32cf7ce1286 100644
--- a/drivers/md/dm-default-key.c
+++ b/drivers/md/dm-default-key.c
@@ -248,6 +248,7 @@ static int default_key_ctr(struct dm_target *ti, unsigned int argc, char **argv)
 	}
 
 	ti->num_flush_bios = 1;
+	ti->flush_bypasses_map = true;
 
 	err = 0;
 	goto out;
-- 
2.49.0

